### PR TITLE
feat(product): add thumbnail field

### PR DESCRIPTION
**Pull request by griest024**
BREAKING CHANGE:
thumbnail is now a required field on DaffProduct.
A number of other previously optional fields are now required.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
- Adds `thumbnail` field
- Makes some optional fields required

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

- Adds `thumbnail` field
- Makes some optional fields required

## Other information

### CODE REVIEW

File: libs/product/src/models/product.ts [hunk 1]
```diff
@@ -16,14 +16,23 @@ export enum DaffProductTypeEnum {
  */
 export interface DaffProduct extends DaffLocatable {
 	id: ID;
-	type?: DaffProductTypeEnum;
+	type: DaffProductTypeEnum;
```
Comment (damienwebdev) [inline libs/product/src/models/product.ts:19]: Can you move the unrelated model changes to another PR?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

**Review by lderrickable — CHANGES_REQUESTED** [review]

File: libs/product/src/models/product.ts (hunk 1 above)
Comment (griest024) [reply libs/product/src/models/product.ts]: removed

File: libs/product/driver/magento/src/transforms/simple-product-transformers.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export function transformMagentoSimpleProduct(product: MagentoProduct, mediaUrl:
     discount: getDiscount(product),
     in_stock: product.stock_status === MagentoProductStockStatusEnum.InStock,
     images: transformMediaGalleryEntries(product, mediaUrl),
+    thumbnail: transformImage(product.image, mediaUrl),
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/driver/magento/src/transforms/simple-product-transformers.ts]: changed

File: libs/product/src/models/product.ts [hunk 3]
```diff
@@ -25,19 +25,35 @@ export interface DaffProduct extends DaffLocatable {
...
-	 */
+  type?: DaffProductTypeEnum;
+  /**
+   * The product name.
+   */
+  name?: string;
+  /**
+   * The main image for the product.
+   */
+  thumbnail: DaffProductImage;
```
Comment (lderrickable) [inline libs/product/src/models/product.ts:36]: I think the thumbnail should be described as "A smaller image to concisely describe the product". Or something like that.

Comment (griest024) [reply libs/product/src/models/product.ts:36]: changed

**Review by lderrickable — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
